src/main/java/org/protege/editor/owl/client/LocalHttpClient.java

```diff
@@ -467,11 +467,12 @@ public SnapShot getSnapShot(@Nonnull ProjectId projectId) throws LoginTimeoutExc
 		ClientRequestException {
 		String requestUrl = PROJECT_SNAPSHOT + "?projectid=" + projectId.get();
 		Response response = get(requestUrl); // send request to server
-		return retrieveDocumentSnapshotFromServerResponse(response);
+		return retrieveDocumentSnapshotFromServerResponse(response, projectId);
 	}
 
-	private SnapShot retrieveDocumentSnapshotFromServerResponse(Response response)
+	private SnapShot retrieveDocumentSnapshotFromServerResponse(Response response, @Nonnull ProjectId projectId)
```

Comment:
The boilerplate is there for 2 reasons:

1) Nonnull allows intellij to try and look for places in the code where a null parameter might be passed in.
2) The null check will give a decent error message if it actually occurs at runtime. (These checks can be removed once we're confident that we really aren't passing nulls)

And we are actually using the projectId. It's just that we were erroneously getting it from the instance variable where it was null.